PyAngle/Angle.py

```diff
@@ -1,373 +1,279 @@
 import math
+from fractions import Fraction
+from typing import Union
+
+input_t = Union[int, float, str, Fraction]
 
 """
 A simple package for angle calculation
 """
 
 
 class Angle:
-    """Class of Mathematic Angle\n
-    > designed especially for angle in the form of DMS(Degree, Minute and Second)\n
-    > immutable object\n
-    `degree` (int): degree of the angle;\n
-    `minute` (int): minute of the angle;\n
-    `second` (float): second of the angle;\n
+    """Class of Mathematic Angle
+    > designed especially for angle in the fmt of DMS(Degree, Minute and Second)
+    > immutable object
+    `deg` (int): deg of the angle;
+    `minute` (int): minute of the angle;
+    `sec` (float): sec of the angle;
     """
 
-    degree = 0.0  # degree(int when output)
-    minute = 0.0  # minute(int when output)
-    second = 0.0  # second(float, default: "{:.2f}")
-
-    __REL_TOL = 0.001  # max relative diff
-    __ABS_TOL = 0.000001  # max absolute diff
+    _deg: Fraction = Fraction(0)
 
-    def __init__(self, degree=None, minute=None, second=None, rad=None, x=None, y=None):
-        """Creator of Angle\n
-        Format Priority:\n
-        1. DMS: degree, minute, second;\n
-        2. Degree: degree;\n
-        3. Rad: rad;\n
-        4. (x, y): x, y\n
-        """
-        # DMS
-        if degree is not None and minute is not None and second is not None:
-            self.__createByDMS(d=degree, m=minute, s=second)
-        # Degree
-        elif degree is not None:
-            self.__createByDegree(degree=degree)
-        # Radian
-        elif rad is not None:
-            self.__createByRadian(rad=rad)
-        # (x, y)
-        elif x is not None and y is not None:
-            self.__createByXY(x=x, y=y)
-        # invalid params
-        else:
-            # TODO: raise Exception
-            raise GeneratorExit
+    def __init__(self, deg: input_t):
+        self._deg = Fraction(deg)
         self.adjust()
 
-    def __createByDMS(self, d, m=0.0, s=0.0):
-        """Private Initiator For Degree, Minute and Second
+    @staticmethod
+    def from_dms(deg: input_t, minute: input_t = 0, sec: input_t = 0) -> 'Angle':
```

Comment:
constrain types within int/float or input checking?

Reply:
does NOT constrain types. this is just a type HINT, and the code currently does not check for unsupported types